source/adapters/opencl/command_buffer.cpp

```diff
@@ -29,13 +70,26 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferCreateExp(
   if (!clCreateCommandBufferKHR || Res != CL_SUCCESS)
     return UR_RESULT_ERROR_INVALID_OPERATION;
 
+  const bool IsUpdatable =
+      pCommandBufferDesc ? pCommandBufferDesc->isUpdatable : false;
+
+  bool SupportsUpdate = false;
+  cl_device_id CLDevice = cl_adapter::cast<cl_device_id>(hDevice);
+  CL_RETURN_ON_FAILURE(
+      deviceSupportsURCommandBufferKernelUpdate(CLDevice, SupportsUpdate));
+
+  const bool Updatable = IsUpdatable && SupportsUpdate;
```

Comment:
Should we return a warning or something if IsUpdatable==true and SupportUpdate==false?